### DISCUSSION TITLE
Jax's internal notion of a function

### DISCUSSION THREAD

**Discussion by pharringtonp19** [jax-ml/jax#13135] [Q&A] [answered]
I was wondering if the following quote taken from [AOT](https://jax.readthedocs.io/en/latest/aot.html#inspecting-staged-out-computations) could be elaborated upon. -- Thanks!

> JAX’s internal notion of a function specialized to the types of its arguments is not always a reified data structure in memory

**Comment by jakevdp** [comment, marked as answer]
I think what that is meant to convey is that in JAX, functions aren't self-contained data structures that can be inspected directly. Rather, functions are represented by arbitrary Python bytecode that must be traced (via `jax.make_jaxpr` or similar) in order to ascertain their effect on any given inputs. I believe @froystig was the author there - he might weigh-in as well.

**Reply by pharringtonp19** [reply]
@jakevdp Thanks for the response
